File: go-client-sdk/pkg/models/shared/taskcreate.go
package shared

type TaskCreate struct {
	AccountID  *string                `json:"accountId,omitempty"`
	Additional map[string]interface{} `json:"additional,omitempty"`
	Body       *string                `json:"body,omitempty"`
	ContactID  *string                `json:"contactId,omitempty"`
	DealID     *string                `json:"dealId,omitempty"`
	DueDate    *string                `json:"dueDate,omitempty"`
	IsDone     *bool                  `json:"isDone,omitempty"`
	LeadID     *string                `json:"leadId,omitempty"`
	Priority   *string                `json:"priority,omitempty"`
	Status     *string                `json:"status,omitempty"`
	Subject    *string                `json:"subject,omitempty"`
	UserID     *string                `json:"userId,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/taskcreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class TaskCreate extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountId" })
  accountId?: string;

  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=body" })
  body?: string;

  @SpeakeasyMetadata({ data: "json, name=contactId" })
  contactId?: string;

  @SpeakeasyMetadata({ data: "json, name=dealId" })
  dealId?: string;

  @SpeakeasyMetadata({ data: "json, name=dueDate" })
  dueDate?: string;

  @SpeakeasyMetadata({ data: "json, name=isDone" })
  isDone?: boolean;

  @SpeakeasyMetadata({ data: "json, name=leadId" })
  leadId?: string;

  @SpeakeasyMetadata({ data: "json, name=priority" })
  priority?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: string;

  @SpeakeasyMetadata({ data: "json, name=subject" })
  subject?: string;

  @SpeakeasyMetadata({ data: "json, name=userId" })
  userId?: string;
}
